nbi/engine/build.xml

```diff
@@ -20,10 +20,17 @@
 
 -->
 
-<project name="NBI_Engine" default="default" basedir=".">
+<project name="nbi/engine" default="default" basedir=".">
```

Comment:
Please don't change the project name. We tried building NB installer for 12.0 after merging your change set, It failed with below error
BUILD FAILED
netbeans/nbbuild/installer/infra/build/build.xml:133: The following error occurred while executing this line:
nbbuild/installer/infra/build/nbi_all/nbi/engine/build.xml:47: Cannot find /Users/sarillamohanarao/Mohan/Code_Repository/Github_Repository/Installer_Maven_Issue/netbeans/nbbuild/installer/infra/build/nbi_all/nbbuild/templates/projectized.xml imported from /Users/sarillamohanarao/Mohan/Code_Repository/Github_Repository/Installer_Maven_Issue/netbeans/nbbuild/installer/infra/build/nbi_all/nbi/engine/build.xml

Reply:
I had to change the project name to get `projectized.xml` to work. This was needed to download the stubs to compile `InitializeMacJDK9` with JDK8. 

The build error above is not due to changing the project name, instead it is due to the fact that the relative path does not work when building the NB installer.

```xml
<import file="../../nbbuild/templates/projectized.xml"/>
```

Do you know how the project can download the stubs `applemenu-external-desktop-classes-8.2.zip` without using `projectized.xml`?

Reply:
@oyarzun , indeed import of projectized.xml is causing the issue. I currently don't have solution for this issue. Is PR is urgent or we can wait for 12.1, as changing something in NB installer script is risky at this moment as we are near to NB 12.0 release.

Reply:
@arusinha & @ebarboni It can wait until 12.1 since this has been broken since version 10. Plus there are ways around it, create one installer for Java 8 and another for Java 11+. 

I do not use this feature and only started looking at it since we switched our RCPs builds to use maven. In maven there is no goal to build a macOS application like there is with ant.

https://issues.apache.org/jira/browse/NETBEANSINFRA-37

Reply:
@arusinha 

I've found a work around that allows the NB installer to be built. Since it is not used for macOS we don't need the mac specific classes that are loaded by reflection. Create a copy of the original nbi/engine build.xml in nbbuild/installer/infra/build/build-nbi-engine.xml then patch it's build.xml as follows.

```diff
diff --git a/nbbuild/installer/infra/build/build.xml b/nbbuild/installer/infra/build/build.xml
index 4861fc7092..94134b2634 100644
--- a/nbbuild/installer/infra/build/build.xml
+++ b/nbbuild/installer/infra/build/build.xml
@@ -71,8 +71,13 @@
                 <exclude name="infra/server/**/build/**/*.*"/>
                 <exclude name="infra/server/**/dist/**/*.*"/>
                 <exclude name="infra/server/**/private/**/*.*"/>
+                <!-- macOS uses native installer so ignore initialize macOS classes loaded with reflection -->
+                <exclude name="engine/src/org/netbeans/installer/wizard/containers/InitializeMacJDK8.java"/>
+                <exclude name="engine/src/org/netbeans/installer/wizard/containers/InitializeMacJDK9.java"/>
             </fileset>
         </copy>
+        <!--replace nbi/engine build.xml with one that does not build the initialize macOS classes -->
+        <copy file="build-nbi-engine.xml" tofile="${nbi.core.dir}/engine/build.xml" overwrite="true"/>
         <copy todir="${nbi.netbeans.dir}">
             <fileset dir="${hg.root}/installer">
                 <exclude name="infra/server/**/build/**/*.*"/>
```

Reply:
@oyarzun , separate NB installer is released each for Linux/Mac/Windows.  
Conditional import can come handy.
Possible solution can be 
<property name="IS_NB_INSTALLER" value="true" />(this property has to be defined in installer/infra/build/build.xml)


nbi/engine/build.xml  changes
<condition property="relative_path" value="../../../../" else="../../">
 <isset property="IS_NB_INSTALLER"/>
        </condition>
            <import file="${relative_path}/nbbuild/templates/projectized.xml"/>

Reply:
@arusinha I like the idea of a property, but I tried changing the relative path. I ran into the same problem with projectized.xml which made me change the project name. The new path causes this check to fail when building the NB installer.

```xml
    <property name="ant.file.1" location="${ant.file}"/>
    <property name="ant.file.2" location="${nb_all}/${ant.project.name}/build.xml"/>
    <fail message="Please keep the Ant project name the same as the project's directory name: ${ant.file.1} vs. ${ant.file.2}">
        <condition>
            <not>
                <equals arg1="${ant.file.1}" arg2="${ant.file.2}"/>
            </not>
        </condition>
    </fail>
```

Reply:
@oyarzun I am working on some other things, will revisit this issue once those are done

Reply:
@arusinha when you have time to circle back to this, I have a new branch that has this fix working without using `projectized.xml`. 

https://github.com/oyarzun/netbeans/commit/cba705bde9c76c0a65cbd5e2dfaac757660d92fe

Let me know if you want me to submit a new PR.

Reply:
John  is now managing mac NB Installer. @mcdonnell-john request you to review the changes.